File: Plugin/plugin-monitor/src/main/java/com/kronos/plugin/monitor/scan/vcs/GitStatusCollege.java
package com.kronos.plugin.monitor.scan.vcs;

import com.kronos.plugin.monitor.repo.data.GitStatus;
import com.kronos.plugin.monitor.utils.FileUtils;

import org.gradle.api.initialization.Settings;
import org.gradle.api.invocation.Gradle;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.function.Consumer;

public class GitStatusCollege {

    public List<GitStatus> execute() {
        List<GitStatus> gitStatuses = new ArrayList<>();
        gitStatuses.add(new GitStatusTask(FileUtils.rootFile.getAbsolutePath(), "root").execute());
        return gitStatuses;
    }

    public static class GitStatusTask {
        public String wp;
        public String path;

        GitStatusTask(String wp, String path) {
            this.wp = wp;
            this.path = path;
        }

        public GitStatus execute() {
            GitStatus gitStatus = new GitStatus();
            gitStatus.path = path;
            if (new File(wp).exists() && new File(wp, ".git").exists()) {
                //     gitStatus.commit = GitTools.getCommit(wp);
                //   gitStatus.status = ProcessWarp.process(GitTools.statusDetail(wp)).text;
                // gitStatus.branch = ProcessWarp.process(GitTools.currentBranch(wp)).text;
            } else {
                //  gitStatus.status = "当前仓库没有权限或没有拉取下来"
            }
            return gitStatus;
        }
    }
}

File: Plugin/plugin-monitor/src/main/java/com/kronos/plugin/monitor/scan/vcs/GitStatusCollege.kt
package com.kronos.plugin.monitor.scan.vcs

import com.kronos.plugin.monitor.utils.GitUtil.getCommit
import com.kronos.plugin.monitor.utils.GitUtil.fileStatus
import com.kronos.plugin.monitor.utils.GitUtil.curBranchName
import com.kronos.plugin.monitor.repo.data.GitStatus
import com.kronos.plugin.monitor.scan.vcs.GitStatusCollege.GitStatusTask
import com.kronos.plugin.monitor.utils.FileUtils
import java.io.File
import java.util.ArrayList

class GitStatusCollege {

    fun execute(): List<GitStatus> {
        val gitStatuses: MutableList<GitStatus> = ArrayList()
        gitStatuses.add(GitStatusTask(FileUtils.rootFile.absolutePath, "root").execute())
        return gitStatuses
    }

    class GitStatusTask internal constructor(var wp: String, var path: String) {
        fun execute(): GitStatus {
            val gitStatus = GitStatus()
            gitStatus.path = path
            val file = File(wp)
            if (file.exists() && File(file, ".git").exists()) {
                gitStatus.commit = getCommit(file)
                gitStatus.status = fileStatus(file)
                gitStatus.branch = curBranchName(file)
            } else {
                gitStatus.status = "当前仓库git有异常"
            }
            return gitStatus
        }
    }
}